MSI-1590: Use Array Key as Sku due to change in array structure

josh-carter opened the pull request:
### Description

Previously the SKU was part of the array data, its now been altered so that the SKU is the key for each array of stock data, currently Source Items are not created for newly created products via import.

### Fixed Issues (if relevant)
1. magento-engcom/msi#1590: After Import of products via Catalog Product Import SourceItems are not created in MSI

### Manual testing scenarios
1. Create CSV with new products in Import format
2. Import Products via Admin
3. Check Source Items are successfully created

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

josh-carter committed:
Use Array Key as Sku due to change in array structure

- Previously the SKU was part of the array data, its not been altered so that the SKU is the key for each array of stock data

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/signed)](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1592) <br/>All committers have signed the CLA.

josh-carter committed:
Alter StockItemImporter Integration to match expected array stucture

josh-carter committed:
Fix type error in GetProductTypesBySkus::getResultKey 

- Previously if an integer SKU like 12345, integer was returned instead of string

josh-carter committed:
Create new Integration test for Product Import

- Tests that when a Product import is performed a Source Item is correctly created/updated

josh-carter committed:
Ensure string version of SKU is passed to LegacyuStock Item setter

- If SKU is numerical then it will be passed as integer
